### PR TITLE
fix: branch creation and misc fixes

### PR DESCRIPTION
**Pull request by pawanpaudel93**
## Summary

This pull request addresses multiple improvements and bug fixes:

- Resolved the `fetchUserRepos` function issue so that repos user is contributor of also shows on the user profile.
- Implemented branch name cleaning using [clean-git-ref](https://github.com/TheSavior/clean-git-ref/) during new branch creation similar to what GitHub has, accompanied by toast error message.
  - Before hitting Create
  image
  - After hitting Create
  image
  - Recreating an existing branch
  image

- Shorten address lengths for reviewers and assignees address in the PR and Issue page.
  image
  image

- Updated the profile username error message for improved clarity.
  image

### CODE REVIEW

File: src/pages/issue/read/components/Sidebar.tsx [hunk 1]
```diff
@@ -25,7 +26,7 @@ export default function Sidebar() {
           selectedIssue.assignees.map((assignee) => (
             <div className="flex items-center gap-1 text-gray-600">
               <PiDotDuotone className="min-w-[24px] h-6 !text-yellow-500" />
-              <span className="truncate">{assignee}</span>
+              <span className="truncate">{shortenAddress(assignee, 6)}</span>
```
Comment (7i7o) [inline src/pages/issue/read/components/Sidebar.tsx:29]: Does it have a way to see the full address? Like a tooltip when you hover for a few seconds?

File: src/pages/pull/read/components/ReviewerAdd.tsx [hunk 2]
```diff
@@ -64,7 +65,7 @@ export default function ReviewerAdd() {
                                 reviewers.indexOf(address) > -1 ? 'opacity-100' : 'opacity-0'
                               }`}
                             />
-                            {address}
+                            {shortenAddress(address, 6)}
```
Comment (7i7o) [inline src/pages/pull/read/components/ReviewerAdd.tsx:68]: Does it have a way to see the full address? Like a tooltip when you hover for a few seconds?

File: src/pages/pull/read/components/Sidebar.tsx [hunk 3]
```diff
@@ -63,7 +64,7 @@ export default function Sidebar() {
               ) : (
                 <PiDotDuotone className="min-w-[24px] h-6 !text-yellow-500" />
               )}
-              <span className="truncate">{reviewer.address}</span>
+              <span className="truncate">{shortenAddress(reviewer.address, 6)}</span>
```
Comment (7i7o) [inline src/pages/pull/read/components/Sidebar.tsx:67]: Does it have a way to see the full address? Like a tooltip when you hover for a few seconds?

**Comment by 7i7o** [comment]
Ignore my previous comments! All good! 🚀

**Review by 7i7o — APPROVED** [review]
LGTM! 🚀

**Review by kranthicodes — APPROVED** [review]
utACK! Thanks for working on it.

**Pull request merged**